packages/js-example-app/src/index.client.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  BlrButtonIcon,
+  BlrButtonText,
+  BlrCheckbox,
+  BlrInputFieldNumber,
+  BlrInputFieldText,
+  BlrRadio,
+  BlrSelect,
+  BlrTabBar,
+  BlrTextarea,
+  BlrToggleSwitch,
+} from '@boiler/ui-library';
+
+await hydrate();
+init();
+
+async function hydrate() {
+  // Start fetching the Lit hydration support module (note the absence
+  // of "await" -- we don't want to block yet).
+  const litHydrateSupportInstalled = import('@lit-labs/ssr-client/lit-element-hydrate-support.js');
+
+  if (!HTMLTemplateElement.prototype.hasOwnProperty('shadowRoot')) {
+    // Fetch the declarative shadow DOM polyfill.
+    const { hydrateShadowRoots } = await import('@webcomponents/template-shadowroot/template-shadowroot.js');
+
+    // Apply the polyfill. This is a one-shot operation, so it is important
+    // it happens after all HTML has been parsed.
+    hydrateShadowRoots(document.body);
+  }
+
+  await litHydrateSupportInstalled;
+
+  // Import component modules causing them to become interactive
+  await import('@boiler/ui-library');
```

Comment:
I have noticed that the component gets loaded visually but the functionality is not there yet ( clicking on 'Toggle loading state' does not do anything), might be that the hydration has not completed yet at this point even though the elements are already painted. In this case, is there a way to wait and show the loading text or something similar while the hydration is still going on.

Reply:
Noted in new [Issue](https://github.com/deven-org/boiler/issues/1122#issue-2348276750). See point **A)**